third_party/amd/lib/TritonAMDGPUTransforms/RescheduleOps.cpp

```diff
@@ -434,13 +522,22 @@ struct TritonAMDGPURescheduleOps
     for (auto it = rescheduledOps.rbegin(); it != rescheduledOps.rend(); ++it) {
       (*it)->moveBefore(mlirBlock, mlirBlock->begin());
     }
+
+    OpBuilder builder(&(mlirBlock->front()));
+    for (auto &op : mlirBlock->getOperations()) {
+      if (dyn_cast<triton::LoadOp>(&op)) {
```

Comment:
Is this placing a sched.bar(0) after every load? Is this meant to be an experiment or it is meant to be performant generally?

Reply:
It is experimental. We can change it. The idea is that we need to prevent the backend to move `global/buffer_loads` around. I noticed that the backend moves all `global_loads` to one place - i.e., right before the second barrier in the case of our gemm kernel